### ISSUE TITLE
Development tools article

### ISSUE THREAD

**Issue by stubailo** [meteor/guide#312]
This would cover just basic setup stuff for newbies, including:
1. Using version control with Meteor (using the default gitignore, etc)
2. Some default suggestions for editors and plugins to use (perhaps atom with eslint integration)
3. JavaScript debugging via console.log, meteor debug, and chrome devtools

**Comment by skirunman** [comment]
[This information](https://github.com/meteor/meteor/blob/devel/docs/client/full-api/concepts.html#L457) was useful to us and is not captured in the Meteor Guide to my knowledge. Could be added to debugging article.

```
When debugging your app, your browser's JavaScript console behaves as
if it were attached to your app's namespace. You see your app's
globals and the exports of the packages that your app uses
directly. You don't see the variables from inside those packages, and
you don't see the exports of your transitive dependencies (packages
that aren't used directly by your app, but that are used by packages
that are used by your app).

If you want to look inside packages from inside your in-browser
debugger, you've got two options:

* Set a breakpoint inside package code. While stopped on that
  breakpoint, the console will be in the package's namespace. You'll
  see the package's package-scope variables, imports, and also any
  file-scope variables for the file you're stopped in.

* If a package `foo` is included in your app, regardless of whether
  your app uses it directly, its exports are available in
  `Package.foo`. For example, if the `email` package is loaded, then
  you can access `Package.email.Email.send` even from namespaces that
  don't use the `email` package directly.
```

**Comment by stubailo** [comment]
Hmm, I think it would need to be updated for the new world with `import`. @benjamn is there a good way to access modules from the browser console?

**Comment by skirunman** [comment]
Agreed. I think same issue of scope from the debug console. Will be undefined unless you are in code where that object has been imported, at least that is how I understand things.

**Comment by tmeasday** [comment]
We should include this content: https://github.com/meteor/meteor/wiki/Tools-for-a-simple-development-environment-for-Meteor-on-Windows

**Comment by benjamn** [comment]
Unfortunately, the semantics of `import` are not really compatible with `meteor shell` or any shell-like REPL environment: https://github.com/meteor/meteor/issues/6271#issuecomment-187999135

Because symbols imported using `import` are mutable bindings, a compiler for `import` statements needs the complete picture: not only what you imported and what module it came from, but where those symbols are used in the rest of your code. A REPL divorces the `import` statement from the symbol usage, making the compiler's job impossible.

In short, when you're using `meteor shell`, you need to use `require`.

**Comment by lorensr** [comment]
+1 for atom. Article should crosslink to http://guide.meteor.com/mobile.html#logging-and-remote-debugging

**Comment by batjko** [comment]
Would we add things like boilerplates, default lint and build configs etc.? In other words, what should be the scope of this tooling article?

**Comment by stubailo** [comment]
I was thinking more like the stuff you use to debug your app, etc. unless the build configs and boilerplates are something you need for every app, it doesn't seem super relevant?

**Comment by batjko** [comment]
@stubailo I agree. Just thinking about what expectations might be with an article headlined "Development Tools"; it's a wide term.

**Comment by filipenevola** [comment]
I'm closing this issue because it's too old.

We are going to make many updates to the guide in the next weeks and it's better to focus on issues with recent activity. If you think this issue is still relevant please open a new one.
